### PR TITLE
HHH-10429 - Change SimpleValue isIdentityColumn method to return true…

### PR DESCRIPTION
**Pull request by yinzara**
… if the generator class used extends from IdentityGenerator instead of simply being equal to IdentityGenerator

Added test case to verify new functionality.
Discovered defect in PurchaseRecordIdGenerator used in CompositeIdWithGeneratorTest as it should implement IdentifierGenerator not extend IdentityGenerator (as it is not an Identity column).
This caused the CompositeIdWithGeneratorTest to fail with the change to SimpleValue

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/test/cid/PurchaseRecordIdGenerator.java [hunk 1]
```diff
@@ -8,14 +8,15 @@
...
 import org.hibernate.id.IdentityGenerator;
 
 /**
- * Simple {@link IdentityGenerator} implementation for testing composite-id.
+ * Simple {@link IdentifierGenerator} implementation for testing composite-id.
  * 
  * @author Jacob Robertson
  */
-public class PurchaseRecordIdGenerator extends IdentityGenerator {
+public class PurchaseRecordIdGenerator implements IdentifierGenerator {
```
Comment (vladmihalcea) [inline hibernate-core/src/test/java/org/hibernate/test/cid/PurchaseRecordIdGenerator.java]: This is not required since `IdentityGenerator` implements `IdentifierGenerator` and it's a proof that the proposed changes don't affect generators that were created prior to this PR.

Comment (yinzara) [inline hibernate-core/src/test/java/org/hibernate/test/cid/PurchaseRecordIdGenerator.java]: Unfortunately I have to disagree with you. I would never have changed the class if the related test case didn't fail.  The problem is this test case is written incorrectly.  The IdGenerator implemented has nothing to do with an IdentityGenerator, it is simply a custom implementation of the IdentifierGenerator interface.  By extending IdentifierGenerator the class implies it somehow relies on the "identity" support by some databases however this generator would work with any database. With the change to SimpleValue in this pull request, all tests in CompositeIdwithGeneratorTest started to fail.  This is because the creation statement for the PurchaseRecord table fails because it says there are two defined primary keys which is caused because one primary key is defined in the PurchaseRecord.hbm.xml and the other being inferred because the first column is now considered an identity column because the generator class incorrectly extends from IdentityGenerator instead of implementing IdentifierGenerator.  The table create statement ends up being:
create table PurchaseRecord (
        purchaseNumber integer generated by default as identity,
        purchaseSequence varchar(255) not null,
        "timestamp" timestamp,
        primary key (purchaseNumber, purchaseSequence)
    )


In short, the only possible defects that could arise in existing applications from this change is if a Hibernate user has defined a customer IdentifierGenerator and extended from IdentityGenerator incorrectly (instead of implementing IdentifierGenerator).  In that case the generated SQL for the database table could incorrectly infer a column as an identity column.  This would ONLY affect HSQL and MySQL since they're the only database types currently supporting this.

**Comment by vladmihalcea** [comment]
Hi @yinzara. Did you sign the [CLA](https://cla.jboss.org/)?

**Comment by vladmihalcea** [comment]
I replaced this [PR](https://github.com/hibernate/hibernate-orm/pull/1309) with this one.
